fix:added a method to identify the data types properly and show in th…

Naveen-Goud opened the pull request:
**Description of PR**
- this pr contains the changes to handle the data types from the user regarding  [feature](#7176 )

**Changes in PR**
- In this PR added a method ,which takes the json data and it identifies the datatypes, then saves in firestore as expected.

fixes- #7176 

**output**
- below are the screen shots , after sending the data, they are stored in firestore with proper datatype format.
![Screenshot from 2024-05-31 16-26-01](https://github.com/appsmithorg/appsmith/assets/72243823/ec142ead-fa18-4d6f-90eb-c22c6ba8d46d)
![Screenshot from 2024-05-31 16-26-52](https://github.com/appsmithorg/appsmith/assets/72243823/a500a565-ee6d-4ff8-a2af-5cefe8e274e2)
![Screenshot from 2024-05-31 16-27-06](https://github.com/appsmithorg/appsmith/assets/72243823/16e20fb2-b0c2-444c-b17a-2a560fd86b37)


below is the screen shot, when we try to get the  documents then they are also represented based on their format
![Screenshot from 2024-05-31 16-28-26](https://github.com/appsmithorg/appsmith/assets/72243823/41a76ac9-11b7-4792-95cd-36e029eb542b)

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit


- **New Features**
	- Enhanced data handling capabilities for geolocation and timestamp data in the Firestore integration.
	- Improved validation for Firestore document references.
	- Added robust data conversion methods for better formatting of Firestore data.

- **Bug Fixes**
	- Fixed issues related to incorrect formatting of timestamp and geolocation data.

- **Tests**
	- Introduced a new test suite for validating the functionality of the Firestore plugin against an emulated Firestore environment.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

sagar-qa007 commented on app/server/appsmith-plugins/firestorePlugin/src/main/java/com/external/plugins/FirestorePlugin.java:
```diff
@@ -829,6 +833,30 @@ private Mono<ActionExecutionResult> methodAddToCollection(
                     });
         }
 
+        private Object checkAndConvertDataType(Object value, CollectionReference collection) {
```
@Naveen-Goud Please add unit test case.

Naveen-Goud replied:
Hii @sagar-qa007 ,I have added the test case for this scenario. Please can you  go through it.

Naveen-Goud commented:
Hii @somangshu @NilanshBansal @Nikhil-Nandagopal , this PR has been approved , can you  merge the changes !!

thank you.

NilanshBansal commented:
@Naveen-Goud this needs further testing from our end. We will be prioritising this in the next week.
cc: @rohan-arthur @carinanfonseca

rishabhrathod01 commented on app/server/appsmith-plugins/firestorePlugin/src/main/java/com/external/plugins/FirestorePlugin.java:
```diff
@@ -25,15 +25,8 @@
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.google.api.core.ApiFuture;
 import com.google.auth.oauth2.GoogleCredentials;
-import com.google.cloud.firestore.CollectionReference;
```
@Naveen-Goud please add specific imports instead of using import *

rishabhrathod01 commented:
@Naveen-Goud the server unit test is failing for above PR, could you please check.

Naveen-Goud commented:
Hii @rishabhrathod01 , I have updated the test files and removed few statement in FirestorePluginTest, since they are covered in my test file.

thank you.

rishabhrathod01 commented:
@Naveen-Goud could you please resolve the merge conflict by taking pull from latest release?

Naveen-Goud commented:
Hii @rishabhrathod01 , I have resolved the  merge conficts , could you review this PR.

thank you.

rishabhrathod01 commented:
@Naveen-Goud there are server unit tests that are failing on the PR, could you please resolve them.

Naveen-Goud commented:
Hi @NilanshBansal ,I have checked the test case failing files, those files are not related to the changes I have made.

thank you.
